server/src/helpers/extrinsics.ts

```diff
@@ -3,6 +3,11 @@ import { AnyTuple } from '@polkadot/types-codec/types';
 
 import { TxMethodLC, TxSectionLC } from '../types';
 
+const TRANSFER_METHODS: string[] = [
+  TxMethodLC.TRANSFER,
+  TxMethodLC.TRANSFER_KEEP_ALIVE,
+  TxMethodLC.TRANSFER_ALLOW_DEATH,
+];
```

Comment:
I see `transferall` [here](https://github.com/gear-foundation/integrations-rosetta-api/pull/155/files#diff-1b87f9d30cd8d855238e19288647b8e081b76416525dc395760183d56726dae3R54), but not listed here. Should we include it?

Reply:
This is because we can't get the amount of transferred tokens [here](https://github.com/gear-foundation/integrations-rosetta-api/blob/673dd5c10c5cb539adcfd5f6bb1f58c2ce9fc692/server/src/helpers/transaction.ts#L41). So if it fails it cannot be included in the transactions list. But everything will be fine if it succeeds because we take data from the `Transfer` event

Reply:
@osipov-mit Could we do something like this then? My concern is we're changing the behavior by dropping this tx from the list if it fails.

Was `transfer_all` part of the rename of the methods?

```
  // Collect transfer operations
  if (isTransferTx(tx) && opStatus === OperationStatus.FAILURE) {
    const src = tx.signer.toJSON()['id'];
    const dest = tx.args[0].toJSON()['id'];
    if (tx.method.toLowerCase() === TxMethodLC.TRANSFER_ALL) {
      const amount = null
    else {
      const amountVal = new BN(tx.args[1].toString());
      const amount = new Amount(amount.toString(), currency);
    }

    if (src && dest) {
      const amount = new BN(tx.args[1].toString());
      operations.push(
        Operation.constructFromObject({
          operation_identifier: new OperationIdentifier(operations.length),
          type: OpType.TRANSFER,
          status: opStatus,
          account: new AccountIdentifier(dest),
          amount: amount,
        }),
      );
```

Reply:
Yes we can do it if that's ok with you.

`transfer_all` isn't part of renaming I just found out that we hadn't tracked this before.